src/module/actor/entity.js

```diff
@@ -434,52 +442,62 @@ export default class OseActor extends Actor {
     }
   }
 
+  /**
+   * Constructs the attack roll formula for an attack
+   *
+   * @param {object} attData - the attack data
+   * @param {OseItem} attData.item - the item being used in the attack
+   * @param {OseActor} attData.actor - this actor (at least it should be!)
+   * @param {object} attData.roll - the attack roll data
+   * @param {string} attData.roll.save - the save roll that may be used against the attack
+   * @param {OseActor} attData.roll.target - the target of the attack
+   * @param {object} options - the type of attack and whether to skip the dialog
+   * @param {string} options.type - the type of attack, i.e. melee or missile
+   * @param {boolean} options.skipDialog - whether to skip the dialog
+   * @returns {OseDice.Roll} - the attack roll with completed rollParts and other data
+   */
   rollAttack(attData, options = {}) {
     const data = this.system;
 
-    const rollParts = ["1d20"];
-    const dmgParts = [];
-    let label = game.i18n.format("OSE.roll.attacks", {
-      name: this.name,
-    });
-    if (attData.item) {
-      label = game.i18n.format("OSE.roll.attacksWith", {
-        name: attData.item.name,
-      });
-      dmgParts.push(attData.item.system.damage);
-    } else {
-      dmgParts.push("1d6");
-    }
+    const label = attData.item
+      ? game.i18n.format("OSE.roll.attacksWith", {
+          name: attData.item.name,
+        })
+      : game.i18n.format("OSE.roll.attacks", {
+          name: this.name,
+        });
+
+    const dmgParts = removeFalsyElements([
+      // Weapon damage roll value
+      attData.item?.system?.damage ?? "1d6",
+      // Weapon damage bonus
+      attData.item?.system?.bonus,
+    ]);
 
+    const rollParts = ["1d20"];
     const ascending = game.settings.get(game.system.id, "ascendingAC");
-    if (ascending) {
-      rollParts.push(data.thac0.bba.toString());
-    }
-    if (options.type === "missile") {
-      rollParts.push(
-        data.scores.dex.mod.toString(),
-        data.thac0.mod.missile.toString()
-      );
-    } else if (options.type === "melee") {
-      rollParts.push(
-        data.scores.str.mod.toString(),
-        data.thac0.mod.melee.toString()
-      );
-    }
-    if (attData.item && attData.item.system.bonus) {
-      rollParts.push(attData.item.system.bonus);
-    }
-    const thac0 = data.thac0.value;
-    if (options.type === "melee") {
-      dmgParts.push(data.scores.str.mod);
+
+    if (ascending && data.thac0.bba) {
```

Comment:
Since the if/else at line 487 doesn't use curlies, should we remove them here, since we're not doing multiple things in this if block?

Reply:
hmmm can I promise to catch it in the #387 PR?

Reply:
Hahaha fine, I'll stop nitpicking. :P

Reply:
this is not likely the only place where we do this, so I think we'll need to enable an optional linting rule